tests/optim/utils/test_model_utils.py

```diff
@@ -0,0 +1,249 @@
+#!/usr/bin/env python3
+import unittest
+
+import torch
+import torch.nn.functional as F
+
+import captum.optim._utils.models as model_utils
+from captum.optim._models.inception_v1 import googlenet
+from tests.helpers.basic import BaseTest, assertTensorAlmostEqual
+
+
+class TestLocalResponseNormLayer(BaseTest):
+    def test_local_response_norm_layer(self) -> None:
+        size = 5
+        alpha = 9.999999747378752e-05
+        beta = 0.75
+        k = 1
+
+        x = torch.randn(32, 5, 24, 24)
+        lrn_layer = model_utils.LocalResponseNormLayer(
+            size=size, alpha=alpha, beta=beta, k=k
+        )
+
+        assertTensorAlmostEqual(
+            self,
+            lrn_layer(x),
+            F.local_response_norm(x, size=size, alpha=alpha, beta=beta, k=k),
+            0,
+        )
+
+
+class TestReluLayer(BaseTest):
+    def test_relu_layer(self) -> None:
+        x = torch.randn(1, 3, 4, 4)
+        relu_layer = model_utils.ReluLayer()
+        assertTensorAlmostEqual(self, relu_layer(x), F.relu(x), 0)
+
+
+class TestRedirectedReluLayer(BaseTest):
+    def test_forward_redirected_relu_layer(self) -> None:
+        x = torch.randn(1, 3, 4, 4)
+        layer = model_utils.RedirectedReluLayer()
+        assertTensorAlmostEqual(self, layer(x), F.relu(x), 0)
+
+    def test_backward_redirected_relu_layer(self) -> None:
+        t_grad_input, t_grad_output = [], []
+
+        def check_grad(self, grad_input, grad_output):
+            t_grad_input.append(grad_input[0].clone().detach())
+            t_grad_output.append(grad_output[0].clone().detach())
+
+        rr_layer = model_utils.RedirectedReluLayer()
+        x = (torch.randn(1, 3, 4, 4, requires_grad=True) - 5).clamp(0, 1)

```

Comment:
I think that instead of using torch.randn here and in line 41 it would be better to use a fixed input. It would be easier to understand and debug the functionality independent of the random seed. 

when I run:  `x = (torch.randn(1, 3, 4, 4, requires_grad=True) - 5).clamp(0, 1)` locally, I see all zero -valued tensor. We could explicitly test for that as a special case.

Reply:
It looks like I was just creating a zeros tensor in an odd way for testing. I replaced it with torch.zeros and it works just as well!